presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java

```diff
@@ -188,29 +192,29 @@ private boolean isAggregationPushdownSupported(AggregationNode partialAggregatio
             return true;
         }
 
-        private boolean isSupportedOrcTypeForMinMax(Type type)
+        private boolean isNotSupportedOrcTypeForMinMax(Type type)
         {
-            return !BOOLEAN.equals(type) &&
-                    type.getJavaType() != boolean.class &&
-                    !isRowType(type) &&
-                    !isArrayType(type) &&
-                    !isMapType(type) &&
-                    !TINYINT.equals(type) &&
-                    !VARBINARY.equals(type) &&
-                    !TIMESTAMP.equals(type) &&
-                    ((!VARBINARY.equals(type) && !VARCHAR.equals(type)) ||
-                            isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session));
+            return BOOLEAN.equals(type) ||
+                    type.getJavaType() == boolean.class ||
+                    isRowType(type) ||
+                    isArrayType(type) ||
+                    isMapType(type) ||
+                    TINYINT.equals(type) ||
+                    VARBINARY.equals(type) ||
```

Comment:
VARBINARY is checked twice, the other check is on the line 205

Reply:
That is correct. Although this duplication is only for the ORC part. Both ORC and Parquet branches can now reuse the newly refactored `isNotSupportedForVariableLengthDataType` method which improves readability overall.

Reply:
I'm confused, having VARBINARY here will satisfy this condition and it will never reach the isNotSupportedForVariableLengthDataType. Imho the VARBINARY type should only be checked in isNotSupportedForVariableLengthDataType.

Btw, Unsupported might sound a bit better than NotSupported.

Reply:
>  Imho the VARBINARY type should only be checked in isNotSupportedForVariableLengthDataType.

If the session property is set as true, then the method will return false. So if VARBINARY check is indeed true and if we are only checking for it inside the method wouldn't the overall logic will fail for ORC.